wordpress.yaml

```diff
@@ -1,7 +1,136 @@
 ---
-- name: Deploy Wordpress
+- name: Setting up wordpress
   hosts: all
   tasks:
+    - name: Update repositories 
+      become: true
+      apt:
+        #upgrade: yes
+        update_cache: yes
+        cache_valid_time: 86400 #A day
+      
+    - name: Install php dependencies
+      become: true
+      apt:
+        name:
+          - php-cli
+          - php-fpm
+          - php-mysql
+          - php-json
+          - php-opcache
+          - php-mbstring
+          - php-xml
+          - php-gd
+          - php-curl
+        state: present
+     
+    - name: Install nginx
+      become: true
+      apt:
+        name: nginx 
+        state: present
 
-    - name: message to remove
-      command: echo 'Here you gonna put all tasks and roles that gonna be runing on the VM'
+    - name: Open firewall for nginx
+      become: true
+      ufw:
+        rule: allow
+        name: 'Nginx Full'
+    
+    - name: Copy NGINX default settings file
+      become: true
+      copy:
+        src: /vagrant/default
+        dest: /etc/nginx/sites-enabled
+        force: yes
+
+    - name: Install MySQL Server
+      become: true
+      apt:
+        name: 
+          - mysql-server
+          - python3-pymysql  
+        state: present
+    
+    # Set MySQL root Password on Ansible/Ubuntu - https://linuxhint.com/set_mysql_root_password_ansible/ 
+    - name: Relax the MySQL root user
+      shell: sudo mysql -u root -e 'UPDATE mysql.user SET plugin="mysql_native_password"
+       WHERE user="root" AND host="localhost"'
+
+    - name: Flush Privileges
+      shell: sudo mysql -u root -e 'FLUSH PRIVILEGES'
+
+    - name: Create SQL user 
+      mysql_user:
+        login_host: 'localhost'
+        login_user: 'root'
+        login_password: ''
+        name: wp_user
+        password: wp_password
+        priv: "*.*:ALL"
+        state: present
+     
+    - name: Create DB for wordpress
+      mysql_db:
+        login_host: 'localhost'
+        login_user: 'wp_user'
+        login_password: 'wp_password'
+        name: wp_db
+        state: present
+    
+    - name: Check if wp-cli exists
+      stat: path="/bin/wp"
+      register: wpcli_exist
+      tags:
+        - always
+
+    - name: updating wp-cli
+      become: true
+      shell: "/bin/wp cli update --yes"
+      when: wpcli_exist.stat.exists
+      tags:
+        - wp
+        - update
+    
+    - name: Download wp-cli
+      become: true
+      get_url:
+        url="https://raw.githubusercontent.com/wp-cli/builds/gh-pages/phar/wp-cli.phar"
+        dest="/bin/wp"
+        force_basic_auth=yes
+        mode=0755
+      when: not wpcli_exist.stat.exists
+      tags:
+        - wp
+        - install  
+
+    - name: Checking if Wordpress is installed
+      stat: path=/var/www/html/wordpress
+      register: wp_installed
+
+    - name: Install Wordpress
+      shell: "sudo wp core download --path=/var/www/html/wordpress --allow-root"
```

Comment:
why `sudo`?

Reply:
also it's always recommend to run commands as non-root user

Reply:
As I'm downloading the wordpress core inside a protected subdirectory, root permissions were required. wp-cli advices against using root and that's why I use the `--allow-root` flag.

I could try to don't use wp-cli as root but I would have to add more tasks.